### ISSUE TITLE
[feature request] Add KAN models

### ISSUE THREAD

**Issue by leo-smi** [keras-team/keras#19914]
Recently was released a paper called: [KAN: Kolmogorov-Arnold Networks](https://arxiv.org/abs/2404.19756)

There's some [repos ](https://github.com/mintisan/awesome-kan)about this topic, but would be nice if Keras has the model by defalt.

I think the current "problem" is about the validation performance on these models. So is there a way to add a playgroud space on keras models? For example: ```keras.applications.playground.KAN```

Thank you for your patient.

**Comment by SamanehSaadat** [comment]
Hi @leo-smi!

According to [Keras API Design Guidelines](https://github.com/keras-team/governance/blob/master/keras_api_design_guidelines.md), to add new features, "they should be widely recognized as a machine learning best practice.  We will not add new layers/etc that were recently published to ArXiv.org, even in case of claims of increased accuracy/etc."

We will consider adding this feature if it stands the test of time.
